src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -469,6 +463,32 @@ protected function initializeBundles()
     }
 
     /**
+     * @param BundleInterface $bundle
+     * @param bool $dependency
+     */
+    protected function addBundle($bundle, $dependency = false)
+    {
+        $name = $bundle->getName();
+
+        if (isset($this->bundles[$name]) && false === $dependency) {
```

Comment:
Hmmmm yeah you're right. I had some kind of plugin-based system in my mind in which we prefer less run-time errors.

But you're right, when we're talking about bundles and their dependencies it's wrong to fail silently, the developer must know the warning.